collector/generatorreceiver/generator_receiver.go

```diff
@@ -209,3 +215,29 @@ func newTraceReceiver(config *Config,
 	genReceiver.traceConsumer = consumer
 	return &genReceiver, nil
 }
+
+func validateConfiguration(topoFile topology.File) error {
+	err := flags.Manager.ValidateFlags()
+	if err != nil {
+		return fmt.Errorf("validation of flag configuration failed: %v", err)
+	}
+
+	for _, service := range topoFile.Topology.Services {
+		err = service.Validate(*topoFile.Topology)
```

Comment:
I don't love how I passed in `Topology` as an argument to `service.Validate()`. However, it's necessary because` service.Validate()` calls `route.validate()`, which checks that the route's downstream calls are all valid. In order to do that, it needs access to `topology` so that it can call `topology.GetServiceTier()` on its downstream calls. Any ideas on improving this?

Reply:
We could add a ServiceManager along the lines of FlagManager. This would simplify a lot of things I think - eg. `TraceGenerator` currently has a `topology` attribute, but it's only used to call `GetServiceTier`.

I think that should probably wait for a later ticket, though, so we can focus on getting the functionality we need for our current goals ASAP.

Reply:
Yes I realized after I wrote this that FlagManager accomplishes just that! But agreed that we should wait for a later ticket :)